File: app/src/types/serviceOffering.ts
export interface ServiceOffering {
  id: number;
  name: string;
  serviceCategoryId: number;
  category: string;
  description?: string | null;
  durationMinutes?: number | null;
  price?: number | null;
  addOnDetails?: string | null;
  addOnRate?: number | null;
  isHomeService: boolean;
  isActive: boolean;
  createdAt: string;
  lastModifiedAt?: string | null;
}

export interface SaveServiceOfferingRequest {
  name: string;
  serviceCategoryId: number;
  category?: string | null;
  description?: string | null;
  durationMinutes?: number | null;
  price?: number | null;
  addOnDetails?: string | null;
  addOnRate?: number | null;
  isHomeService: boolean;
  isActive: boolean;
}

File: backend/src/Sixram.Nextgen.JinmarxWellness.Domain/Entities/ServiceOffering.cs
using Sixram.Nextgen.JinmarxWellness.Domain.Common;

namespace Sixram.Nextgen.JinmarxWellness.Domain.Entities;

public class ServiceOffering : AuditableEntity
{
    public string Name { get; set; } = string.Empty;
    public int ServiceCategoryId { get; set; }
    public ServiceCategory ServiceCategory { get; set; } = null!;
    public string Category { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int? DurationMinutes { get; set; }
    public decimal? Price { get; set; }
    public string? AddOnDetails { get; set; }
    public decimal? AddOnRate { get; set; }
    public bool IsHomeService { get; set; }
    public bool IsActive { get; set; } = true;
}
